### PR TITLE
[Doc] Add documentation for remote cluster connection mode for CCR

### PR DESCRIPTION
**Pull request by skumarp7**
### Description
This PR involves addition of documentation for remote cluster connection modes for CCR

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks so much for creating this PR, @skumarp7! I will start the review process.

**Comment by nisgoel-amazon** [comment]
@ankitkala can you take a look

**Review by ankitkala — APPROVED** [review]
LGTM!

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 1]
```diff
@@ -145,6 +145,12 @@ docker inspect --format='{% raw %}{{range .NetworkSettings.Networks}}{{.IPAddres
 
 Cross-cluster replication follows a "pull" model, so most changes occur on the follower cluster, not the leader cluster. 
 
+### Connection modes to a remote cluster
+
+- Sniff Mode:
```
Comment (vagimeli) [inline _tuning-your-cluster/replication-plugin/getting-started.md:150]:
```suggestion
The connection modes include _sniff mode_ and _proxy mode_.
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 2]
```diff
@@ -162,6 +168,26 @@ curl -XPUT -k -H 'Content-Type: application/json' -u 'admin:<custom-admin-passwo
 }'
 ```
 
+- Proxy Mode:
```
Comment (vagimeli) [inline _tuning-your-cluster/replication-plugin/getting-started.md:171]:
```suggestion
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 3]
```diff
@@ -145,6 +145,12 @@ docker inspect --format='{% raw %}{{range .NetworkSettings.Networks}}{{.IPAddres
 
 Cross-cluster replication follows a "pull" model, so most changes occur on the follower cluster, not the leader cluster. 
 
+### Connection modes to a remote cluster
+
+- Sniff Mode:
+
+In sniff mode, a remote connection is created in the follower cluster by specifying a name and a list of seed nodes of the leader cluster. While establishing the connection, the leader's cluster state is fetched from one of its seed nodes. This mode mandates that the publish addresses of the seed nodes must be reachable from the follower cluster. Sniff mode is the default connection mode.
```
Comment (vagimeli) [inline _tuning-your-cluster/replication-plugin/getting-started.md:152]:
```suggestion
In sniff mode, the follower cluster establishes a remote connection to the leader cluster by specifying a name and a list of seed nodes from the leader cluster. During the connection setup, the follower cluster retrieves the leader cluster's state from one of the provided seed nodes. This mode requires that the publish addresses of the seed nodes in the leader cluster are accessible from the follower cluster. Sniff mode is the default connection mode.
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 4]
```diff
@@ -162,6 +168,26 @@ curl -XPUT -k -H 'Content-Type: application/json' -u 'admin:<custom-admin-passwo
 }'
 ```
 
+- Proxy Mode:
+
+In proxy mode, a remote connection is created in the follower cluster by specifying a name and a single proxy address of the leader cluster. While establishing the connection, a configurable number of socket connections are opened up to the proxy address. The responsibility of the proxy is to direct these connections to the leader cluster. Unlike other connection modes, proxy mode does not necessitate that leader cluster nodes have accessible publish addresses.
```
Comment (vagimeli) [inline _tuning-your-cluster/replication-plugin/getting-started.md:173]:
```suggestion
In proxy mode, the follower cluster establishes a remote connection to the leader cluster by specifying a name and a single proxy address. During the connection setup, a configurable number of socket connections are opened to the provided proxy address. The proxy's responsibility is to direct these connections to the appropriate nodes in the leader cluster. Unlike other connection modes, proxy mode does not require the nodes in the leader cluster to have publicly accessible publish addresses. 
```

**Review by vagimeli — COMMENTED** [review]
Doc review is complete. Once the edits are reviewed and accepted, I'll move this forward to editorial review. Thank you, Melissa

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 5]
```diff
@@ -145,6 +145,12 @@ docker inspect --format='{% raw %}{{range .NetworkSettings.Networks}}{{.IPAddres
 
 Cross-cluster replication follows a "pull" model, so most changes occur on the follower cluster, not the leader cluster. 
 
+### Connection modes to a remote cluster
```
Comment (natebower) [inline _tuning-your-cluster/replication-plugin/getting-started.md:148]:
```suggestion
### Connection modes for a remote cluster
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 6]
```diff
@@ -145,6 +145,12 @@ docker inspect --format='{% raw %}{{range .NetworkSettings.Networks}}{{.IPAddres
...
+
+The connection modes include _sniff mode_ and _proxy mode_.
+
+In sniff mode, the follower cluster establishes a remote connection to the leader cluster by specifying a name and a list of seed nodes from the leader cluster. During the connection setup, the follower cluster retrieves the leader cluster's state from one of the provided seed nodes. This mode requires that the publish addresses of the seed nodes in the leader cluster are accessible from the follower cluster. Sniff mode is the default connection mode.
```
Comment (natebower) [inline _tuning-your-cluster/replication-plugin/getting-started.md:152]:
```suggestion
In sniff mode, the follower cluster establishes a remote connection to the leader cluster by specifying a name and a list of seed nodes from the leader cluster. During the connection setup, the follower cluster retrieves the leader cluster's state from one of the provided seed nodes. This mode requires the publish addresses of the seed nodes in the leader cluster to be accessible from the follower cluster. Sniff mode is the default connection mode.
```

File: _tuning-your-cluster/replication-plugin/getting-started.md [hunk 7]
```diff
@@ -162,6 +168,25 @@ curl -XPUT -k -H 'Content-Type: application/json' -u 'admin:<custom-admin-passwo
 }'
 ```
 
+
+In proxy mode, the follower cluster establishes a remote connection to the leader cluster by specifying a name and a single proxy address. During the connection setup, a configurable number of socket connections are opened to the provided proxy address. The proxy's responsibility is to direct these connections to the appropriate nodes in the leader cluster. Unlike other connection modes, proxy mode does not require the nodes in the leader cluster to have publicly accessible publish addresses.
```
Comment (natebower) [inline _tuning-your-cluster/replication-plugin/getting-started.md:172]:
```suggestion
In proxy mode, the follower cluster establishes a remote connection to the leader cluster by specifying a name and a single proxy address. During the connection setup, a configurable number of socket connections to the provided proxy address are opened. The proxy's responsibility is to direct these connections to the appropriate nodes in the leader cluster. Unlike other connection modes, proxy mode does not require the nodes in the leader cluster to have publicly accessible publish addresses:
```

**Review by natebower — COMMENTED** [review]
@skumarp7 @vagimeli Please see my changes and let me know if you have any questions. Thanks!

**Review by vagimeli — APPROVED** [review]
Addressed editorial feedback. Permission to push line 172 change is blocked. Will open new PR to make that change.

**Pull request merged**